test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/KieServerWebSocketScalingIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+
+public class KieServerWebSocketScalingIntegrationTest {
+
+    private GenericScenario workbenchMonitoringScenario;
+    private GenericScenario kieServerScenario;
+
+    private WorkbenchDeployment workbenchDeployment;
+    private KieServerDeployment kieServerDeployment;
+
+    private KieServerMgmtControllerClient kieControllerClient;
+    private KieServicesClient kieServerClient;
+
+    private static final String CONTAINER_ID = "cont-id";
+    private static final String CONTAINER_ALIAS = "cont-alias";
+
+    private static final String PROJECT_GROUP_ID = "org.kie.server.testing";
+    private static final String DEFINITION_PROJECT_SNAPSHOT_NAME = "definition-project-snapshot";
+    private static final String DEFINITION_PROJECT_SNAPSHOT_VERSION = "1.0.0-SNAPSHOT";
+
+    private static final String WEBSOCKET_CONNECTION = "Connection to Kie Controller over websocket is now open";
+    private static final String STARTED_CONTAINER = "Container cont-id (for release id org.kie.server.testing:definition-project-snapshot:1.0.0-SNAPSHOT) successfully started";
+
+    @Before
+    public void setUp() {
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project-snapshot").getFile());
+
+        DeploymentScenarioBuilderFactory deploymentScenarioFactory = DeploymentScenarioBuilderFactoryLoader.getInstance();
+
+        DeploymentSettings workbenchMonitoringSettings = deploymentScenarioFactory.getWorkbenchMonitoringSettingsBuilder()
+                .withControllerUser(DeploymentConstants.getControllerUser(), DeploymentConstants.getControllerPassword())
+                .build();
+        workbenchMonitoringScenario = deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withMonitoring(workbenchMonitoringSettings)
+                .build();
+        workbenchMonitoringScenario.deploy();
+        workbenchDeployment = workbenchMonitoringScenario.getWorkbenchDeployments().get(0);
+
+        DeploymentSettings kieServerSettings = deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+                .withControllerUser(DeploymentConstants.getControllerUser(), DeploymentConstants.getControllerPassword())
+                .withControllerProtocol(workbenchDeployment.getWebSocketUri().getScheme())
+                .withControllerConnection(workbenchDeployment.getWebSocketUri().getHost(), String.valueOf(workbenchDeployment.getWebSocketUri().getPort()))
+                .withMavenRepoUrl(MavenConstants.getMavenRepoUrl())
+                .withMavenRepoUser(MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .withKieServerSyncDeploy(true)
+                .build();
+        kieServerScenario = deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withKieServer(kieServerSettings)
+                .build();
+        kieServerScenario.deploy();
```

Comment:
Is there any reason for 2 scenarios? We can have Kie server settings and Workbench setting in one scenario.

Reply:
Yes, two scenarios were chosen to verify that WebSocket correctly works for cases when Kie servers are placed in different project than controller.